search/utils.py

```diff
@@ -72,13 +69,24 @@ def lower_string(self):
         return self._lower.isoformat()
 
 
-def doc_type_warning_decorator(func):
+# FIXME: delete this method when ES index doc_type parameter
+#  will be removed from the edx-platform
+def doc_type_warning(func):
     """
-    Decorator which mark function with doc_type as deprecated.
+    Raise deprecation warning.
+
+    Decorator which mark function with doc_type parameter as deprecated.
+    This method needs for compatibility with interface that still use
+    doc_types and should be removed with doc_types from this lib.
     """
     @functools.wraps(func)
-    def wrapper(ref, *args, **kwargs):
+    def wrapper(self_or_cls, *args, **kwargs):
         if kwargs.get("doc_type"):
```

Comment:
Do we need the `doc_type` param is in the kwargs? If not `pop` could be used.

Reply:
We don't use `doc_type`. Done